### PR TITLE
build: clean up project configuration

### LINKED ISSUE #39: build: move project metadata to pyproject.toml
As part of this issue, we need to deprecate the use of `setup.cfg` and `setup.py` for project metadata and setup and move all the configuration to `pyproject.toml`

### PR DESCRIPTION
**Pull request by alohamora**
## Description

Move config for the following tools to `pyproject.toml`
1. mypy
2. coverage
3. flake8

Move project metadata defined in `setup.py` to `pyproject.toml`
Fixes #39 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist:

- [ ] My code follows the [style guidelines](https://github.com/elixir-cloud-aai/elixir-cloud-aai/blob/dev/resources/contributing_guidelines.md#language-specific-guidelines) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have not reduced the existing code coverage
- [ ] I have added docstrings following the [Python style guidelines](https://github.com/elixir-cloud-aai/elixir-cloud-aai/blob/dev/resources/python.md) of this project to all new modules, classes, methods and functions are documented with docstrings following; I have updated any previously existing docstrings, if applicable
- [ ] I have updated any sections of the app's documentation that are affected by the proposed changes, if applicable

### CODE REVIEW

File: cloud_registry/version
Comment (alohamora) [inline cloud_registry/version]: had to create a separate version text file because pyproject.toml config does not support reading version from a python variable.
we will have to either define the version in `pyproject.toml` itself or define it in a separate file as above

File: cloud_registry/version.py
Comment (alohamora) [inline cloud_registry/version.py:1]: deleted this file because pyproject config does not support reading version from python variable

File: setup.cfg
Comment (alohamora) [inline setup.cfg]: flake8 config could not be moved to pyproject.toml since flake8 doesn't support reading config from the file.
open issue - https://github.com/PyCQA/flake8/issues/234
to use pyproject.toml based config for flake8, we can use this library - https://pypi.org/project/Flake8-pyproject/

File: cloud_registry/version
Comment (kushagra189) [reply cloud_registry/version]: I might not be aware. Need pointers to why are we using `pyproject.toml` instead of setup config? Is there any deprecation coming around or already published?

File: setup.cfg
Comment (kushagra189) [reply setup.cfg]: Should we consider including it in this PR?

Comment (kushagra189) [reply setup.cfg:2]: Should we consider including it in this PR?

Comment (kushagra189) [reply setup.cfg]: Lastly are we still using this fine anywhere?

**Review by kushagra189 — COMMENTED** [review]
Require some context here for `pyproject.toml` context. Is this specific to cloud-registry or can we export it using foca and use it across other repositories? Want the reasoning for this migration for my understanding.

File: cloud_registry/version
Comment (alohamora) [reply cloud_registry/version]: yes, as per PEP 621 standard, python community has decided to use pyproject.toml as the main source of configuration for the build process.
https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html

File: setup.cfg
Comment (alohamora) [reply setup.cfg]: yes, we are using this file for flake8 config which is run during github ci/cd

**Comment by alohamora** [comment]
@uniqueg can you please review this PR

File: cloud_registry/version
Comment (uniqueg) [reply cloud_registry/version]: If you can define it in `pyproject.toml` itself, then great. If we need it elsewhere, it shouldn't be too hard to parse it from there (at least not compared to a text file)

File: docs/api/conf.py
Comment (uniqueg) [inline docs/api/conf.py]: I've recently used markdown-based API reference docs and find them more satisfying than this stuff + Read the Docs, in many ways.

Of course, for now update the version parsing. Just wanted to give a heads up that we'll probably replace this with a different mechanism soon.

File: setup.cfg
Comment (uniqueg) [reply setup.cfg]: Yes, I'm aware that flake8 is holding off on `pyproject.toml` adoption. However, did you try out this package? Does it work? I'd kinda like to have everything in the same file. But of course it's also not worth going through hoops for that. So if it works fine, cool, let's use it for now (until official support) - otherwise let's keep `setup.cfg` for now.

**Review by uniqueg — APPROVED** [review]
Thanks a lot. LGTM. Happy to merge it as is, but if you like, you can update the version and/or use the drop in package for Flake8 `pyproject.toml` support (see comments). Up to you.

**Comment by alohamora** [comment]
yes, I have tried out the package. It works fine with no changes required in the config format. will integrate both of the above mentioned changes in the PR

File: requirements_dev.txt
Comment (alohamora) [inline requirements_dev.txt:3]: had to upgrade flake8 so that it is compatible with the `flake8-pyproject` package

File: requirements.txt
Comment (alohamora) [inline requirements.txt:1]: connexion released a new major version which has breaking changes. so had to restrict the connexion package limit to `<3.0.0`

**Comment by alohamora** [comment]
@uniqueg can you please have a look at the PR again, there have been dependency changes due to connexion's new release as well.

File: requirements_dev.txt
Comment (uniqueg) [reply requirements_dev.txt:3]: Fair enough, it was quite old :)

File: requirements.txt
Comment (uniqueg) [reply requirements.txt:1]: Can you please raise an issue for that in FOCA? Guess we should upgrade it at some point (but maybe not just right away...)

Comment (uniqueg) [reply requirements.txt:1]: Then after you can resolve this conversation and merge the PR :)

**Review by uniqueg — APPROVED** [review]

File: requirements.txt
Comment (alohamora) [reply requirements.txt:1]: raised an issue in foca - https://github.com/elixir-cloud-aai/foca/issues/182

**Pull request merged**
